src/components/common/ConflictResolver.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Checkbox,
+  FormControlLabel,
+  makeStyles,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+} from '@material-ui/core';
+import Check from '@material-ui/icons/Check';
+import DoneAllIcon from '@material-ui/icons/DoneAll';
+import Error from '@material-ui/icons/Error';
+import ExpandMore from '@material-ui/icons/ExpandMore';
+import clsx from 'clsx';
+import React, { useEffect } from 'react';
+
+import {
+  ConflictResolutionStrategy,
+  QuestionComparisonStatus,
+} from 'generated/sdk';
+
+const useStyles = makeStyles((theme) => ({
+  heading: {
+    fontWeight: 'bold',
+  },
+  icon: {
+    marginRight: theme.spacing(2),
+  },
+  error: {
+    color: theme.palette.error.main,
+  },
+  resolved: {
+    color: '#ffbb00',
+  },
+  check: {
+    color: '#00ff00',
+  },
+  highlight: {
+    backgroundColor: '#faed27',
+  },
+}));
+
+export function ConflictResolver<T>(props: {
+  comparison: T;
+  onConflictResolved: (
+    comparison: T,
+    resolution: ConflictResolutionStrategy
+  ) => void;
+  getStatus: (comparison: T) => QuestionComparisonStatus;
+  getItemId: (comparison: T) => string;
+  getItemTitle: (comparison: T) => string;
+  getDiffInfo: (comparison: T) => {
+    heading: string;
+    existingVal: React.ReactNode;
+    newVal: React.ReactNode;
+    isDifferent: boolean;
+  }[];
+}) {
+  const { comparison, onConflictResolved } = props;
+  const { getStatus, getItemId, getItemTitle, getDiffInfo } = props;
+
+  const [existingItemCheck, setExistingItemCheck] = React.useState(
+    getStatus(comparison) === QuestionComparisonStatus.SAME
+  );
+  const [newItemCheck, setNewItemCheck] = React.useState(
+    getStatus(comparison) === QuestionComparisonStatus.NEW
+  );
+
+  const classes = useStyles();
+
+  const getStatusIcon = (status: QuestionComparisonStatus) => {
+    const isResolved = existingItemCheck || newItemCheck;
+    switch (status) {
+      case QuestionComparisonStatus.DIFFERENT:
+        return isResolved ? (
+          <Check
+            className={clsx(classes.icon, classes.resolved)}
+            data-cy="resolved-icon"
+          />
+        ) : (
+          <Error
+            className={clsx(classes.icon, classes.error)}
+            data-cy="conflict-icon"
+          />
+        );
+      case QuestionComparisonStatus.NEW:
+        return (
+          <Check
+            className={clsx(classes.icon, classes.check)}
+            data-cy="same-icon"
+          />
+        );
+      case QuestionComparisonStatus.SAME:
+        return (
+          <DoneAllIcon
+            className={clsx(classes.icon, classes.check)}
+            data-cy="same-icon"
```

Comment:
I see that `DoneAllIcon` and the previous `Check` icon have the same `data-cy` attribute. Maybe this is fine but I think it is better to be different.